src/VisualStudio/Core/Def/LanguageClient/LogHubLspLogger.cs

```diff
@@ -66,11 +68,13 @@ public override void LogException(Exception exception, string? message = null, p
         _traceSource.TraceEvent(TraceEventType.Error, id: 0, "Exception: {0}", exception);
     }
 
+    [Obsolete($"Use {nameof(CreateContext)} instead.")]
     public override void LogStartContext(string message, params object[] @params)
     {
         _traceSource.TraceEvent(TraceEventType.Start, id: 0, message);
```

Comment:
Do LogHub logs from VS look the same with this change? I'd worry that without the `Start` and `Stop` events being logged, everything would be in one long stream. Right now its nicely broken up by message, which I _think_ is due to this:

![image](https://github.com/user-attachments/assets/4f20e6ac-0079-4638-877c-815da86347c6)

Reply:
So the start/end methods were pretty broken here because they could start and end on any thread. Unless this was using the correlation ID that comes from here?

https://github.com/dotnet/roslyn/blob/3e2d9723ff35e6868b35885bafd777677d0fb0ce/src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs?plain=1#L180C9-L180C116

I don't understand where that would come from in the LSP case though in VS Code; was your screenshot from VS for Windows?

@dibarbet you know a bit more here?

Reply:
It's definitely less than perfect, and some things escape out of their nice containers into a general log item. My worry is this will cause that to happen for everything.

AFAIK the correctly comes from StreamJsonRpc, setup via this bit of magic: https://github.com/dotnet/roslyn/blob/85d110d5e8a3afef2d9e42560cb3be4890955416/src/VisualStudio/Core/Def/LanguageClient/VisualStudioLogHubLoggerFactory.cs#L60

Reply:
I'm not 100% sure but I don't think the start/stop is necessary on our end.  There's actually two getting output in our loghub logs, I think the second one is us and not needed - 
![image](https://github.com/user-attachments/assets/a33b179a-82ac-4fb1-b222-52d8d9770ecb)

Yes, the `Trace.CorrelationManager.ActivityId` is how loghub logs get correlated in VS (set in the queue after we de-queue) - https://github.com/dotnet/roslyn/blob/3e2d9723ff35e6868b35885bafd777677d0fb0ce/src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs?plain=1#L239

Reply:
I believe streamjsonrpc is logging the other start/stops, so ours isn't that useful (other than indicating when the request actually was de-queued).

Reply:
@dibarbet So StreamJsonRpc is magically setting that prior to invoking us, and we're continuing to pass it around?

Reply:
> @dibarbet So StreamJsonRpc is magically setting that prior to invoking us, and we're continuing to pass it around?

IIRC yes, but it has been a while

Reply:
Still seems to be working for me:

![image](https://github.com/user-attachments/assets/edb69def-75dc-4bfa-a957-e32093c4ee88)

Reply:
Thank you for checking <3